frontend/packages/enduser-frontend/src/components/common/file-upload.vue

```diff
@@ -0,0 +1,212 @@
+        this.onDelete(file)
+      },
+      fileIcon(file) {
+        switch (file.file.type) {
+          case 'image/jpeg':
+          case 'image/png':
+            return 'file-image'
+          case 'application/pdf':
+            return 'file-pdf'
+          case 'application/msword':
+          case 'application/vnd.openxmlformats-officedocument.wordprocessingml.document':
+          case 'application/vnd.oasis.opendocument.text':
+            return 'file-word'
+          default:
+            return 'file'
+        }
+      },
+      progressBarStyles(file) {
+        return `width: ${file.progress}%;`
+      }
+    }
+  }
+</script>
+
+<style lang="scss">
+  .file-upload-container {
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+
+    input {
+      width: 300px;
+    }
+
+    .spacer {
+      padding: 20px;
+    }
+
+    .files {
+      display: flex;
+      flex-direction: column;
+
+      > *:not(:last-child) {
+        margin-bottom: 20px;
+      }
+
+      .file {
+        display: flex;
+        flex-direction: row;
+        flex-wrap: nowrap;
+        color: $grey-dark;
+
+        .file-icon-container {
+          margin-right: 16px;
+          flex: 0;
+        }
+
+        .file-details {
+          flex: 1;
+          display: flex;
+          flex-direction: column;
+          flex-wrap: nowrap;
+
+          .file-header {
+            display: flex;
+            flex-direction: row;
+            flex-wrap: nowrap;
+            justify-content: space-between;
+            font-size: 15px;
+
+            .file-header-icon-button {
+              border: none;
+              background: none;
+              padding: 4px;
+              margin-left: 12px;
+              color: $grey;
+              cursor: pointer;
+
+              &:hover {
+                color: $blue;
+              }
+
+              &:disabled {
+                color: $grey-light;
+                cursor: not-allowed;
+              }
+            }
+
+            span {
+              width: min(400px, 50vw);
+              white-space: nowrap;
+              overflow: hidden;
+              text-overflow: ellipsis;
+            }
+          }
+
+          .file-progress-bar {
+            margin-top: 16px;
+
+            .file-progress-bar-background {
+              position: relative;
+              background: $grey;
+              height: 3px;
+              border-radius: 1px;
+
+              .file-progress-bar-progress {
+                position: absolute;
+                top: 0;
+                left: 0;
+                background: $blue;
+                height: 3px;
+                border-radius: 1px;
+                transition: width 0.5s ease-out;
+              }
+            }
+
+            .file-progress-bar-details {
+              display: flex;
+              flex-direction: row;
+              justify-content: space-between;
+              margin-top: 4px;
+              font-size: 14px;
```

Comment:
Where are these pixel values from? Or what I'm actually after: would it make sense to derive these some other variables (or alternatively from eachother if they are related to one another)?

Reply:
Yes, I guess there are some values that could be bound as variables. Most are not related to each other, though.